hyperspy/tests/mva/test_bss.py

```diff
@@ -122,12 +122,12 @@ def test_mask_diff_order_0_on_loadings(self):
     def test_mask_diff_order_1_on_loadings(self):
         # This test, unlike most other tests, either passes or raises an error.
         # It is designed to test if the mask is correctly dilated inside the
-        # `blind_source_separation_method`. If the mask is not correctely
+        # `blind_source_separation_method`. If the mask is not correctly
         # dilated the nan in the loadings should raise an error.
         mask = self.s._get_navigation_signal(dtype="bool")
         mask.isig[5] = True
         self.s.learning_results.loadings[5, :] = np.nan
-        self.s.blind_source_separation(3, diff_order=1, mask=mask,
+        self.s.blind_source_separation(2, diff_order=1, mask=mask,
```

Comment:
While this is true, it is very inconvenient not to be able to perform BSS on lazy signals. We could instead compute the factors used for BSS, perform BSS and store the results in dask arrays to make `get_bss_model` (the potentially memory intensive step) work lazily.

Reply:
Do you mean that the BSS does not need to be compute lazily, since the data have already been reduced? If so, it should have been already working, including `get_bss_model`!
It should be a bit better now and also better tested.